packages/expo-cli/src/commands/upload/submission-service/android/ServiceAccountSource.ts

```diff
@@ -0,0 +1,73 @@
+import prompt from '../../../../prompt';
+import { existingFile } from '../../../../validators';
+import log from '../../../../log';
+
+enum ServiceAccountSourceType {
+  path,
+  prompt,
+  // credentialsService,
+  // ...
+}
+
+interface ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType;
+}
+
+interface ServiceAccountPathSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.path;
+  path: string;
+}
+
+interface ServiceAccountPromptSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.prompt;
+}
+
+export type ServiceAccountSource = ServiceAccountPathSource | ServiceAccountPromptSource;
+
+async function getServiceAccountAsync(source: ServiceAccountSource): Promise<string> {
+  switch (source.sourceType) {
+    case ServiceAccountSourceType.path:
+      return await handlePathSourceAsync(source);
+    case ServiceAccountSourceType.prompt:
+      return await handlePromptSourceAsync(source);
+  }
+}
+
+async function handlePathSourceAsync(source: ServiceAccountPathSource): Promise<string> {
+  if (!(await existingFile(source.path))) {
+    log.warn(`File ${source.path} doesn't exist.`);
+    return await getServiceAccountAsync({ sourceType: ServiceAccountSourceType.prompt });
+  }
+  return source.path;
+}
+
+async function handlePromptSourceAsync(_source: ServiceAccountPromptSource): Promise<string> {
+  const path = await askForServiceAccountPathAsync();
+  return await getServiceAccountAsync({
+    sourceType: ServiceAccountSourceType.path,
+    path,
+  });
+}
+
+async function askForServiceAccountPathAsync(): Promise<string> {
+  log(
```

Comment:
i had some feedback on this log message but decided it was easier to do it in the form of a pr to your branch: https://github.com/expo/expo-cli/pull/2165

Reply:
Thanks! I merged your branch here.